x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/query/query_tab_unified_components.test.tsx

```diff
@@ -774,95 +774,427 @@ describe('query tab with unified timeline', () => {
+            });
+
+            fireEvent.click(screen.getByTestId('timeline-notes-button-small'));
+
+            await waitFor(() => {
+              expect(screen.getByTestId('add-note-container')).toBeVisible();
+            });
+          },
+          SPECIAL_TEST_TIMEOUT
+        );
+
+        it(
+          'should be cancel adding notes',
+          async () => {
+            renderTestComponents();
+            expect(await screen.findByTestId('discoverDocTable')).toBeVisible();
+
+            await waitFor(() => {
+              expect(screen.getByTestId('timeline-notes-button-small')).not.toBeDisabled();
+            });
+
+            fireEvent.click(screen.getByTestId('timeline-notes-button-small'));
+
+            await waitFor(() => {
+              expect(screen.getByTestId('add-note-container')).toBeVisible();
+            });
+
+            userEvent.type(screen.getByTestId('euiMarkdownEditorTextArea'), 'Test Note 1');
+
+            expect(screen.getByTestId('cancel')).not.toBeDisabled();
+
+            fireEvent.click(screen.getByTestId('cancel'));
+
+            await waitFor(() => {
+              expect(screen.queryByTestId('add-note-container')).not.toBeInTheDocument();
+            });
+          },
+          SPECIAL_TEST_TIMEOUT
+        );
+      });
+    });
+  });
+
+  describe('Leading actions - pin', () => {
+    describe('securitySolutionNotesEnabled = true', () => {
+      beforeEach(() => {
+        (useIsExperimentalFeatureEnabled as jest.Mock).mockImplementation(
+          jest.fn((feature: keyof ExperimentalFeatures) => {
+            if (feature === 'unifiedComponentsInTimelineEnabled') {
+              return true;
+            }
+            if (feature === 'securitySolutionNotesEnabled') {
+              return true;
+            }
+            return allowedExperimentalValues[feature];
+          })
+        );
+      });
+      it(
+        'should have the pin button with correct tooltip',
+        async () => {
+          renderTestComponents();
+
+          expect(await screen.findByTestId('discoverDocTable')).toBeVisible();
+
+          expect(screen.getAllByTestId('pin')).toHaveLength(1);
+          // disabled because it is already pinned
+          expect(screen.getByTestId('pin')).toBeDisabled();
+
+          fireEvent.mouseOver(screen.getByTestId('pin'));
+
+          await waitFor(() => {
+            expect(screen.getByTestId('timeline-action-pin-tool-tip')).toBeVisible();
+            expect(screen.getByTestId('timeline-action-pin-tool-tip')).toHaveTextContent(
+              'This event cannot be unpinned because it has notes'
+            );
+            /*
+             * Above event is alert and not an event but `getEventType` in
+             *x-pack/plugins/security_solution/public/timelines/components/timeline/body/helpers.tsx
+             * returns it has event and not an alert even though, it has event.kind as signal.
+             * Need to see if it is okay
+             *
```

Comment:
@michaelolo24, need your opinion here. What do you think.

Reply:
Hey, it's not ideal, but I'm not going to ask you to rebuild to update the mock to match the 8.0+ data since that might have unintended consequences elsewhere. Also, the `getEventType` helper could probably also check for the `signal` case, but once again...no idea what unintended consequences would happen elsewhere. Thanks for the comment though, will help if/when we get to updating all of this!

Reply:
>I'm not going to ask you to rebuild to update the mock to match the 8.0+ data

I thought about doing that but that does not help us much in test scenarios.

>Also, the getEventType helper could probably also check for the signal case

that is the most strange thing. I think we should correct this function and then one of the test will fail. For now it is okay.